edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs

```diff
@@ -309,6 +306,37 @@ private async Task ManageDeployments(V1ServiceList currentServices, V1Deployment
                         Events.CreateService(s.Metadata.Name);
                     }
                 });
+
+            // remove the old
+            await Task.WhenAll(servicesRemoved.Select(
+                i =>
+                {
+                    Events.DeletingService(i);
+                    return this.client.DeleteNamespacedServiceAsync(i.Metadata.Name, this.k8sNamespace, new V1DeleteOptions());
+                }));
+
+            // Create the new.
+            await Task.WhenAll(newServices.Select(
+                s =>
+                {
+                    Events.CreatingService(s);
+                    return this.client.CreateNamespacedServiceAsync(s, this.k8sNamespace);
+                }));
+        }
+
+        async Task ManageDeployments(V1DeploymentList currentDeployments, List<V1Deployment> desiredDeployments)
+        {
+            Dictionary<string, string> currentDeploymentsFromAnnotations = this.GetCurrentDeploymentConfig(currentDeployments)
+                .ToDictionary(
+                    pair => pair.Key,
+                    pair => pair.Value);
+
+            var deploymentsRemoved = new List<V1Deployment>(currentDeployments.Items);
+            deploymentsRemoved.RemoveAll(
+                d =>
+                {
+                    return desiredDeployments.Exists(i => string.Equals(i.Metadata.Name, d.Metadata.Name));
```

Comment:
in this line `i` and `d` make me read the whole function to read multiple times.
```
return desiredDeployments.Exists(desired => desired.Metadata.Name == removing.Metadata.Name);
```
In my opinion, this looks better